ComponentKitTests/CKComponentActionTests.mm

```diff
@@ -0,0 +1,151 @@
+
+  CKComponentActionSend(@selector(testAction:context:), innerComponent, context);
+
+  XCTAssert(actionContext == context, @"Context should match what was passed to CKComponentActionSend");
+
+  [mountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+- (void)testSendActionStartingAtSenderNextResponderReachesParentComponent
+{
+  __block BOOL outerReceivedTestAction = NO;
+  __block BOOL innerReceivedTestAction = NO;
+
+  CKTestActionComponent *innerComponent =
+  [CKTestActionComponent
+   newWithBlock:^(CKComponent *sender, id context){ innerReceivedTestAction = YES; }
+   component:[CKComponent new]];
+
+  CKTestActionComponent *outerComponent =
+  [CKTestActionComponent
+   newWithBlock:^(CKComponent *sender, id context){ outerReceivedTestAction = YES; }
+   component:innerComponent];
+
+  // Must be mounted to send actions:
+  UIView *rootView = [UIView new];
+  NSSet *mountedComponents = CKMountComponentLayout([outerComponent layoutThatFits:{} parentSize:{}], rootView);
+
+  CKComponentActionSend(@selector(testAction:context:), innerComponent);
+
+  XCTAssertTrue(outerReceivedTestAction, @"Outer component should have received action sent by inner component");
+  XCTAssertFalse(innerReceivedTestAction, @"Inner component should not have received action sent from it");
+
+  [mountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+- (void)testSendActionStartingAtSenderDoesNotReachParentComponent
+{
+  __block BOOL outerReceivedTestAction = NO;
+  __block BOOL innerReceivedTestAction = NO;
+
+  CKTestActionComponent *innerComponent =
+  [CKTestActionComponent
+   newWithBlock:^(CKComponent *sender, id context){ innerReceivedTestAction = YES; }
+   component:[CKComponent new]];
+
+  CKTestActionComponent *outerComponent =
+  [CKTestActionComponent
+   newWithBlock:^(CKComponent *sender, id context){ outerReceivedTestAction = YES; }
+   component:innerComponent];
+
+  // Must be mounted to send actions:
+  UIView *rootView = [UIView new];
+  NSSet *mountedComponents = CKMountComponentLayout([outerComponent layoutThatFits:{} parentSize:{}], rootView);
+
+  CKComponentActionSend(@selector(testAction:context:), innerComponent, nil, CKComponentActionSendBehaviorStartAtSender);
+
+  XCTAssertFalse(outerReceivedTestAction, @"Outer component should not have received action since inner component did");
+  XCTAssertTrue(innerReceivedTestAction, @"Inner component should have received action");
+
+  [mountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+@end
+
+@implementation CKTestActionComponent
+{
+  void (^_block)(CKComponent *, id);
+}
+
++ (instancetype)newWithBlock:(void (^)(CKComponent *sender, id context))block component:(CKComponent *)component
+{
+  CKTestActionComponent *c = [super newWithComponent:component];
+  if (c) {
+    c->_block = block;
+  }
+  return c;
+}
+
+- (void)testAction:(CKComponent *)sender context:(id)context
+{
+  _block(sender, context);
+}
+
+@end
```

Comment:
Maybe DRY up this "test action component"? Seems like you've written it twice, in both CKComponentActionAttributeTests and CKComponentActionTests.